tmail-backend/rate-limiter/rate-limiter-cassandra/src/main/scala/com/linagora/tmail/rate/limiter/cassandra/CassandraRateLimitPlanTable.scala

```diff
@@ -0,0 +1,34 @@
+package com.linagora.tmail.rate.limiter.cassandra
+
+import com.datastax.driver.core.DataType.{bigint, frozenMap, text, uuid}
+import com.datastax.driver.core.schemabuilder.{Create, SchemaBuilder}
+import org.apache.james.backends.cassandra.components.CassandraModule
+
+object CassandraRateLimitPlanTable {
+  val TABLE_NAME: String = "rate_limit_plan"
+  val PLAN_ID: String = "plan_id"
+  val PLAN_NAME: String = "plan_name"
+  val OPERATION_LIMITATION_NAME: String = "operation_limitation_name";
+  val RATE_LIMITATION_BASE: String = "rate_limitation_base"
+  val RATE_LIMITATION_MAPPING: String = "rate_limitation_mapping"
+
+  object RateLimitationBase {
+    val RATE_LIMITATION_NAME: String = "rate_limitation_name"
+    val PERIOD: String = "period"
+    val LIMIT_MAPPING: String = "limit_mapping"
+  }
+
+  val MODULE: CassandraModule = CassandraModule.builder()
+    .`type`(RATE_LIMITATION_BASE)
+    .statement(statement => statement
+      .addColumn(RateLimitationBase.PERIOD, bigint())
+      .addColumn(RateLimitationBase.LIMIT_MAPPING, frozenMap(text(), bigint())))
+    .table(TABLE_NAME)
+    .comment("Hold Rate Limiting Plan - Use to management.")
+    .statement((statement: Create) => statement
+      .addPartitionKey(PLAN_ID, uuid)
+      .addClusteringColumn(PLAN_NAME, text)
+      .addClusteringColumn(OPERATION_LIMITATION_NAME, text)
+      .addUDTMapColumn(RATE_LIMITATION_MAPPING, text(), SchemaBuilder.frozen(RATE_LIMITATION_BASE)))
```

Comment:
We should likely find a way to freeze this UDT Map too (and not just the UDTs it contains...)

Reply:
Eg in CassandraMailQueueViewModule

```
.addColumn(EnqueuedMailsTable.ATTRIBUTES, frozenMap(text(), blob()))
```

(Where blob is a frozen UDT)

If required we could switch UDTs to tuple types.

Eg:

```
.addColumn(EnqueuedMailsTable.PER_RECIPIENT_SPECIFIC_HEADERS, frozenList(TupleType.of(ProtocolVersion.NEWEST_SUPPORTED, CodecRegistry.DEFAULT_INSTANCE, text(), text(), text()))))
```

(good luck!)

Reply:
> .addColumn(EnqueuedMailsTable.ATTRIBUTES, frozenMap(text(), blob()))

frozenMap not accept UDTType

> If required we could switch UDTs to tuple types.

```
      .addColumn(RATE_LIMITATIONS,
        frozenList(TupleType.of(ProtocolVersion.NEWEST_SUPPORTED, CodecRegistry.DEFAULT_INSTANCE,
          text(),
          bigint(),
          DataType.map(text, bigint())))))
```

it looks good?

Reply:
Better but should'nt we also freeze the map in the tuple?

Reply:
```      .addColumn(RATE_LIMITATIONS,
        frozenList(TupleType.of(ProtocolVersion.NEWEST_SUPPORTED, CodecRegistry.DEFAULT_INSTANCE,
          text(),
          bigint(),
          DataType.frozenMap(text, bigint())))))

```

Reply:
Yes please ensure everything is frozen